### PR TITLE
Feature/metrics

### PR DESCRIPTION
**Pull request by makkalot**

### CODE REVIEW

File: examples/stream/main.go [hunk 1]
```diff
@@ -21,13 +29,22 @@ func main() {
...
 	f.RegisterListHandler("/types/beamery/application", func(event *stream.FlowEvent) error {
 		fmt.Println("got a key from key handler : ", event.Kv.Key)
 		listItems++
 		if listItems%30 == 0 {
-			log.Println("total keys fetched so far : ", listItems)
+			log.Println("total list keys fetched so far : ", listItems)
+		}
+		return nil
+	})
+	f.RegisterWatchHandler("/types/beamery/application", func(event *stream.FlowEvent) error {
```
Comment (AlexsJones) [inline examples/stream/main.go]: maybe use something more open source friendly?

File: examples/stream/main.go [hunk 2]
```diff
@@ -21,13 +29,22 @@ func main() {
 	}
 
 	listItems := 0
+	watchItems := 0
 
 	f := stream.NewEtcdFlow(cli)
 	f.RegisterListHandler("/types/beamery/application", func(event *stream.FlowEvent) error {
```
Comment (AlexsJones) [inline examples/stream/main.go]: Maybe something that would allow someone to run this?

File: examples/stream/main.go (hunk 1 above)
Comment (makkalot) [reply examples/stream/main.go]: yup definitely 👍 🤦‍♂️

File: examples/stream/main.go (hunk 2 above)
Comment (makkalot) [reply examples/stream/main.go]: 👍

**Pull request merged**